app/Http/Controllers/WorkflowController.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\EvidencioAPI;
+use App\Workflow;
+use App\LoadedEvidencioModel;
+use App\Step;
+use App\Field;
+use App\Option;
+use Illuminate\Support\Facades\Gate;
+use Illuminate\Validation\UnauthorizedException;
+
+/**
+ * WorkflowController class, handles database- and API-calls for Workflowpage.
+ */
+class WorkflowController extends Controller
+{
+
+    /**
+     * Show the application dashboard.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index($workflowId)
+    {
+        $result = WorkflowController::loadWorkflow($workflowId);
+        return view('workflow')->with('result', $result);
+    }
+
+    /**
+     * Loads a workflow from the database based on the workflowId
+     *
+     * @param Number $workflowId
+     * @return Array
+     */
+    public function loadWorkflow($workflowId)
+    {
+        $retObj = [];
+        $usedVariables = [];
+        $workflow = Workflow::find($workflowId);
+
+        if (!$workflow->is_verified || !$workflow->is_published) {
+            throw new UnauthorizedException(_("This action is unauthorized."));
+        }
+
+        if ($workflow == null) {
+            $retObj['success'] = false;
+            return $retObj;
+        }
+
+        $retObj['title'] = $workflow->title;
+        $retObj['description'] = $workflow->description;
+        $retObj['languageCode'] = $workflow->language_code;
+        $retObj['evidencioModels'] = $this->getLoadedEvidencioModels($workflow);
+
+        $retObj['steps'] = [];
+        $counter = 0;
+        $steps = $workflow->steps()->get();
+        //dd($steps);
+        foreach ($steps as $step) {
+            $stepLoaded = $this->loadStep($step, $counter, $usedVariables);
+            //dd($stepLoaded);
+            $retObj['steps'][$counter] = $stepLoaded['step'];
+
+            $counter++;
+        }
+
+
+        return $retObj;
+    }
+
+    /**
+     * Returns the IDs of the loaded Evidencio Models of the Workflow
+     *
+     * @param App|Workflow $workflow Workflow to get loaded Evidencio Model IDs from.
+     * @return Array
+     */
+    private function getLoadedEvidencioModels($workflow)
+    {
+        $array = [];
+        $models = $workflow->loadedEvidencioModels()->get();
+        foreach ($models as $model)
+            $array[] = $model->model_id;
+        return $array;
+    }
+
+    /**
+     * Loads the relevant information for a step (NOT including the variables)
+     *
+     * @param App|Step $dbStep Database Model of Step
+     * @param Number $stepNum Number that indicates what step in workflow it is.
+     * @return Array Array containing information of step and the used variables.
+     */
+    private function loadStep($dbStep, $stepNum)
+    {
+        $retObj = [];
+        $retObj['id'] = $dbStep->id;
```

Comment:
You do these types of constructs a lot. There is a lot of duplication here. Try to abstract this to some basic method.